pkg/noun/jets/e/json_de.c

```diff
@@ -0,0 +1,261 @@
+/// @file
+
+#include "jets/k.h"
+#include "jets/q.h"
+#include "jets/w.h"
+
+#include "noun.h"
+
+#include "pdjson.h"
+
+/*
+** custom code for interfacing with external parser:
+** https://github.com/skeeto/pdjson
+*/
+
+typedef struct _u3qedj_coll {
+  c3_b    key_b : 1;  // key flag
+  u3_noun col;        // collection (list for array, map for object)
+  union {             // store context for recursive arrays/objects:
+    u3_noun* tel;     //   - pointer to tail for array
+    u3_atom  key;     //   - key for object
+  };
+} u3qedj_coll;
+
+static u3qedj_coll*
+_push_stack(const u3a_pile *pil_u)
+{
+  u3qedj_coll *res_u = u3a_push(pil_u);
+  u3a_pile_done(pil_u);
+
+  res_u->col = u3_nul;
+  res_u->key = u3_none;
+
+  return res_u;
+}
+
+static u3qedj_coll*
+_pop_stack(const u3a_pile *pil_u)
+{
+  return u3a_pop(pil_u);
+}
+
+static void
+_close_stack(const u3a_pile *pil_u)
+{
+  while ( c3n == u3a_pile_done(pil_u) ) {
+    u3qedj_coll *tak_u = u3a_peek(pil_u);
+
+    u3z(tak_u->col);
+    if ( tak_u->key_b ) {
+      u3z(tak_u->key);
+    }
```

Comment:
The approach here is fine. But it looks like you could also just check that `u3_none != tak_u->key` here, and remove `tak_u->key_b`, since it doesn't otherwise affect control flow.